core/common/src/main/java/alluxio/underfs/UnderFileSystemFactoryRegistry.java

```diff
@@ -115,6 +116,16 @@ public static List<UnderFileSystemFactory> findAll(String path,
             StringUtils.join(supportedVersions, ","), path,
             configuredVersion);
       }
+    } else if (ufsConf.getBoolean(PropertyKey.UNDERFS_STRICT_VERSION_MATCH_ENABLED)
+        && !eligibleFactories.isEmpty()
+        && ufsConf.isSet(PropertyKey.UNDERFS_VERSION)) {
+      String configuredVersion = ufsConf.getString(PropertyKey.UNDERFS_VERSION);
+      Iterator<UnderFileSystemFactory> it = eligibleFactories.iterator();
+      while (it.hasNext()) {
+        if (!configuredVersion.equals(it.next().getVersion())) {
+          it.remove();
+        }
+      }
```

Comment:
An alternative is to change the `supportsPath()` to force a full string match instead of a prefix match now.
https://github.com/Alluxio/alluxio/blob/54b3dc7f8d68b2c90d6dfcfc624419c1bb62d2ed/core/common/src/main/java/alluxio/extensions/ExtensionFactory.java#L41
Which one do you think makes more sense @dbw9580 @maobaolong @Jackson-Wang-7 ?

Reply:
it should retrieve the version from existing eligibleFactories, so I think it's good now.

Reply:
OK let's do the 2nd level filtering here instead of overriding every `supportsPath()`.